group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfigTest.java

```diff
@@ -16,47 +16,118 @@
+        GroupCoordinatorConfig.GROUP_COORDINATOR_CONFIG_DEF,
+        GroupCoordinatorConfig.NEW_GROUP_CONFIG_DEF,
+        GroupCoordinatorConfig.OFFSET_MANAGEMENT_CONFIG_DEF,
+        GroupCoordinatorConfig.CONSUMER_GROUP_CONFIG_DEF,
+        GroupCoordinatorConfig.SHARE_GROUP_CONFIG_DEF
+    );
+
+    public static class CustomAssignor implements ConsumerGroupPartitionAssignor, Configurable {
+        public Map<String, ?> configs;
+
+        @Override
+        public void configure(Map<String, ?> configs) {
+            this.configs = configs;
+        }
+
+        @Override
+        public String name() {
+            return "CustomAssignor";
+        }
+
+        @Override
+        public GroupAssignment assign(
+            GroupSpec groupSpec,
+            SubscribedTopicDescriber subscribedTopicDescriber
+        ) throws PartitionAssignorException {
+            return null;
+        }
+    }
 
     @Test
-    public void testConsumerGroupAssignorsDefault() {
+    public void testConsumerGroupAssignorFullClassNames() {
         // The full class name of the assignors is part of our public api. Hence,
         // we should ensure that they are not changed by mistake.
         assertEquals(
-            List.of(
-                "org.apache.kafka.coordinator.group.assignor.UniformAssignor",
-                "org.apache.kafka.coordinator.group.assignor.RangeAssignor"
-            ),
-            GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_DEFAULT
+            "org.apache.kafka.coordinator.group.assignor.UniformAssignor",
+            UniformAssignor.class.getName()
+        );
+        assertEquals(
+            "org.apache.kafka.coordinator.group.assignor.RangeAssignor",
+            RangeAssignor.class.getName()
         );
     }
 
+    @Test
+    public void testConsumerGroupAssignors() {
+        Map<String, Object> configs = new HashMap<>();
+        GroupCoordinatorConfig config;
+        List<ConsumerGroupPartitionAssignor> assignors;
+
+        // Test short names.
+        configs.put(GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_CONFIG, "range, uniform");
+        config = createConfig(configs);
+        assignors = config.consumerGroupAssignors();
+        assertEquals(2, assignors.size());
+        assertTrue(assignors.get(0) instanceof RangeAssignor);
+        assertTrue(assignors.get(1) instanceof UniformAssignor);
+
+        // Test custom assignor.
+        configs.put(GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_CONFIG, CustomAssignor.class.getName());
+        config = createConfig(configs);
+        assignors = config.consumerGroupAssignors();
+        assertEquals(1, assignors.size());
+        assertTrue(assignors.get(0) instanceof CustomAssignor);
+        assertNotNull(((CustomAssignor) assignors.get(0)).configs);
+
+        // Test with classes.
+        configs.put(GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_CONFIG, List.of(RangeAssignor.class, CustomAssignor.class));
```

Comment:
Should we add test case for a mix of both? something like

List.of("range", CustomAssignor.class)

for completeness, can we add simple failure scenarios as well

Reply:
sg. also extended negative cases in testInvalidConfigs.